### PR TITLE
[#214] Create tcg_rim_tools project

### LINKED ISSUE #214: Import SwidTags project
Import the codebase for creating, processing, and storing swidtags.

### PR DESCRIPTION
**Pull request by chubtub**
Created the codebase for creating and processing swidtags.  It currently supports the following operations:
- creation
- parsing
- validation
Creation can be done with or without csv file input for the <Payload> element.

Closes #214

### CODE REVIEW

File: ACA_SwidTag/src/main/java/hirs/swid/SwidTagGateway.java [hunk 1]
```diff
@@ -0,0 +1,553 @@
...
+            Directory directory = createDirectory(properties);
+            hirs.swid.xjc.File file1 = createFile("Example.com.iotBase.bin", "01.00", "15400");
+            hirs.swid.xjc.File file2 = createFile("iotExec.bin", "01.00", "1024");
+            directory.getDirectoryOrFile().add(file1);
+            directory.getDirectoryOrFile().add(file2);
+            payload.getDirectoryOrFileOrProcess().add(directory);
+            JAXBElement<ResourceCollection> jaxbPayload = objectFactory.createSoftwareIdentityPayload(payload);
+            swidTag.getEntityOrEvidenceOrLink().add(jaxbPayload);
+
+//            JAXBElement<SignatureType> swidtagSignature = objectFactory.createSignature(createSignature());
```
Comment (busaboy1340) [inline ACA_SwidTag/src/main/java/hirs/swid/SwidTagGateway.java]: Still need this code?

File: ACA_SwidTag/src/main/java/hirs/swid/SwidTagGateway.java [hunk 2]
```diff
@@ -0,0 +1,553 @@
...
+        String[] roles = properties.getProperty(SwidTagConstants.ENTITY_ROLE).split(",");
+        for (int i = 0; i < roles.length; i++) {
+            entity.getRole().add(roles[i]);
+        }
+        entity.setThumbprint(properties.getProperty(SwidTagConstants.ENTITY_THUMBPRINT));
+
+        return entity;
+    }
+
+    private Link createLink(Properties properties) {
```
Comment (busaboy1340) [inline ACA_SwidTag/src/main/java/hirs/swid/SwidTagGateway.java]: Need some javadoc for this method?

File: ACA_SwidTag/src/main/java/hirs/swid/SwidTagGateway.java [hunk 3]
```diff
@@ -0,0 +1,553 @@
...
+        Link link = objectFactory.createLink();
+        link.setHref(properties.getProperty(SwidTagConstants.LINK_HREF));
+        link.setRel(properties.getProperty(SwidTagConstants.LINK_REL));
+
+        return link;
+    }
+    private SoftwareMeta createSoftwareMeta(Properties properties) {
```
Comment (busaboy1340) [inline ACA_SwidTag/src/main/java/hirs/swid/SwidTagGateway.java]: Need some javadoc for this method?

File: ACA_SwidTag/src/main/java/hirs/swid/SwidTagGateway.java [hunk 4]
```diff
@@ -0,0 +1,553 @@
...
+        return createFile(file.getName(), "01.00", Long.toString(file.length()));
+    }
+
+    /**
+     *
+     * @param filename
+     * @param location
+     * @return hirs.swid.xjc.File object from File object
+     */
+    private hirs.swid.xjc.File createFile(String filename, String version, String size) {
```
Comment (busaboy1340) [inline ACA_SwidTag/src/main/java/hirs/swid/SwidTagGateway.java]: General comment: Update method descriptions in this file.

File: settings.gradle [hunk 5]
```diff
@@ -6,4 +6,5 @@ include 'TPM_Utils',
         'HIRS_Structs',
         'HIRS_AttestationCA',
         'HIRS_AttestationCAPortal',
+        'ACA_SwidTag',
```
Comment (busaboy1340) [inline settings.gradle]: Do we need to add this sub-project to .travis.yml?

Comment (busaboy1340) [reply settings.gradle]: Tested adding this subproject to Travis. Looks good to add it, if/when we need to.
https://travis-ci.org/nsacyber/HIRS/builds/632261783

**Comment by iadgovuser26** [comment]
Let hold off on putting the docs out there until its ready for use.

**Pull request merged**